### ISSUE TITLE
[FEATURE] Generic way of rendering and interacting with different types of content

### ISSUE THREAD

**Issue by hansmbakker** [mono/SkiaSharp#1486]
# Introduction and use case
This feature request is a continuation of https://github.com/mono/SkiaSharp/issues/1462.
I would like to create a "Composition control" that allows users to make a composition of different layers that are rendered on top of each other:

![image](https://user-images.githubusercontent.com/3463496/91884176-e0877280-ec85-11ea-8331-3a8e0ff75f3a.png)

Suggested interaction:
- one item can be added per layer
- an item can be text, a bitmap or a vector image
- the user can rotate, pan or scale the item in the active layer using touch
- the user can pan the item in the active layer using mouse / pen
- on the left, a list of layers is shown
- the user can select the active layer in the list on the left

For this, it would be helpful if there was a generalized way to render those different types and interact with them (scale/pan/rotate).

What did I investigate:
- we looked at generalization of "renderables" (text/vector/bitmap) in https://github.com/mono/SkiaSharp/issues/1462
- I found [`TouchManipulationBitmap`](https://github.com/xamarin/xamarin-forms-samples/blob/master/SkiaSharpForms/Demos/Demos/SkiaSharpFormsDemos/Transforms/TouchManipulationBitmap.cs) which works for bitmaps, but not for text / paths
- I built a prototype of the control

**Comment by guzuomuse** [comment]
hi @hansmbakker 
do you mean it should be similar like [fabric.js](http://fabricjs.com/) ?
i think the "change layers orders" should be included.
have you got a solution for these features?

**Comment by hansmbakker** [comment]
Yes, it seems like I meant to have a simple version of what fabric.js offers.

Changing the order of layers sounds useful, too.

I only have the prototype I linked, and hoped to get feedback or to start a collaboration because I am a novice with SkiaSharp, but I guess other things have a higher priority, as you are the first commenter on this issue :)

Personally I have laid it to rest for now, but I would welcome others to continue the prototype I started, and maybe you can check whether @mattleibow has an interest for this?
